File: content/snippets/go/design_patterns/class_adapter.go
package main

import "fmt"

// Go doesn't have class inheritance; demonstrate embedding to adapt

type Adaptee struct{}

func (Adaptee) Specific() { fmt.Println("Adaptee specific") }

type Target interface{ Request() }

type ClassAdapter struct{ Adaptee }

func (c ClassAdapter) Request() { c.Specific() }
func main()                     { var t Target = ClassAdapter{}; t.Request() }

File: content/snippets/cpp/design_patterns/class_adapter.cpp
#include <iostream>

class Target { public: virtual void request() = 0; };
class Adaptee { public: void specific() { std::cout<<"Adaptee specific\n"; } };
// C++ class adapter via inheritance isn't common; emulate by inheriting Adaptee and adapting
class ClassAdapter : public Adaptee, public Target { public: void request() override { specific(); } };
int main(){ ClassAdapter a; a.request(); }
